File: AlgoExpert/Sorting/Easy/Insertion Sort.java
import java.util.*;

class Program {
	public static int[] insertionSort(int[] array) {
		// Write your code here.

		for (int i = 0; i < array.length; i++) {
			int j = i;
			while (j > 0 && array[j] < array[j - 1]) {
				swap(j, j - 1, array);
				j--;
			}
		}

		return array;
	}

	public static void swap(int i, int j, int[] array) {
		int temp = array[i];
		array[i] = array[j];
		array[j] = temp;
	}
}

File: Python/AlgoExpert/Sorting/Easy/insertionSort.py
def insertionSort(array):
    # Write your code here.
    for i in range(len(array)):
        j = i
        while j > 0 and array[j] < array[j - 1]:
            swap(array, j, j - 1)
            j -= 1
    return array


def swap(array, i, j):
    array[i], array[j] = array[j], array[i]
